app/models/ldap.rb

```diff
@@ -5,14 +5,25 @@
 class Ldap
   class << self
     def find_by_netid(net_id, ldap_connection: default_connection)
+      attempts ||= 1
+
       filter = Net::LDAP::Filter.eq('uid', net_id)
       result = ldap_connection.search(filter: filter).first
       return {} if result.blank?
 
       attributes(result)
+    rescue Net::LDAP::Error => e
+      if (attempts += 1) < 5 # go back to begin block if condition ok
```

Comment:
@maxkadel just to be certain you are trying to retry a maximum of 3 times?

First pass the attempts will be 2 in the check, then 3, and 4.